docs/design-v1.md

```diff
@@ -0,0 +1,106 @@
+# Caching Service Design Doc
+
+## Purpose
+
+Provide a generalized cache store that can be used from any other service, allowing users to avoid re-running expensive computation that has already been completed for the same dataset.
+
+## MVP Specifications
+
+**_Set a cache value_**
+
+To set a cache value, send a POST request with content-type "multipart/form-data" (to facilitate the file upload).
+
+The fields are:
+
+* `service` - required - name of the service
+* `token` - required - authentication token for the service
```

Comment:
service and token are redundant - the token's username is the service name

at least that's how I assume it'd work. If not there has to be some way of negotiating which service gets which namespace